app/DynamicsAdapter/Fams3Adapter.Dynamics/WebRequestExceptionTranslator.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Newtonsoft.Json;
+using Simple.OData.Client;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Fams3Adapter.Dynamics
+{
+    public class WebRequestExceptionTranslator 
+    {
+        public static string DUPLICATED_ERROR_CODE = "0x80060892";
+        public static string GetErrorCode(WebRequestException webRequestException)
```

Comment:
It could be cleaner if this was `IsDuplicateEntityError` and even better if it was an extension method.

```
public static class WebRequestExceptionExtensions
{
  /* <summary>Determines if the WebRequestException represents a duplicated entity by id or alternate key.</summary>
  public static bool IsDuplicateEntityError(this WebRequestException exception)
  {
    if (e.Code == HttpStatusCode.PreconditionFailed ...
      ...

  }
}

catch (WebRequestException e) when (e.IsDuplicateEntityError()) 
{
}
```

Reply:
right, will change that.